### PR TITLE
Doc py_function unknown shape

### PR DESCRIPTION
**Pull request by bhack**
Try to lower the rate of some recurrent tickets.
See https://github.com/tensorflow/tensorflow/issues/44970#issuecomment-729959298

### CODE REVIEW

File: tensorflow/python/ops/script_ops.py [hunk 1]
```diff
@@ -545,7 +545,31 @@ def my_func(x):
...
+    E.g.
+    ```python
+    import tensorflow.compat.v2 as tf
+    import numpy as np
+
+    #make magical data that requires python
+    def make_data(i): 
+        return np.cast[np.uint8](i) * np.ones([20,256,256,3], dtype=np.float32) / 10.
+
+    #clean up magical data
```
Comment (jsimsa) [inline tensorflow/python/ops/script_ops.py:562]: remove

**Review by jsimsa — CHANGES_REQUESTED** [review]
Thank you. A couple of minor comments.

File: tensorflow/python/ops/script_ops.py [hunk 2]
```diff
@@ -546,20 +546,20 @@ def my_func(x):
...
-    #make magical data that requires python
-    def make_data(i): 
-        return np.cast[np.uint8](i) * np.ones([20,256,256,3], dtype=np.float32) / 10.
+    def make_synthetic_data(i):
+        return np.cast[np.uint8](i) * np.ones([20,256,256,3], 
+                dtype=np.float32) / 10.
 
-    #clean up magical data
     def make_clean_data(i): 
         ones = tf.py_function(make_data,[i],tf.float32)
```
Comment (jsimsa) [inline tensorflow/python/ops/script_ops.py:564]: s/make_data/make_synthetic_data

File: tensorflow/python/ops/script_ops.py [hunk 3]
```diff
@@ -546,20 +546,20 @@ def my_func(x):
...
 
-    #make magical data that requires python
-    def make_data(i): 
-        return np.cast[np.uint8](i) * np.ones([20,256,256,3], dtype=np.float32) / 10.
+    def make_synthetic_data(i):
+        return np.cast[np.uint8](i) * np.ones([20,256,256,3], 
+                dtype=np.float32) / 10.
 
-    #clean up magical data
     def make_clean_data(i):
```
Comment (jsimsa) [inline tensorflow/python/ops/script_ops.py:563]: minor: I would prefer to rename `make_clean_data` with `process_fn`

Comment (bhack) [reply tensorflow/python/ops/script_ops.py]: You are too punctilious towards yourself in the past :stuck_out_tongue_winking_eye:

**Review by jsimsa — APPROVED** [review]

**Pull request merged**